homeassistant/components/tplink/switch.py

```diff
@@ -122,8 +123,23 @@ def update(self):
                 self._mac = self.smartplug.mac
                 self._alias = self.smartplug.alias
                 self._model = self.smartplug.model
-
-            self._state = self.smartplug.state == self.smartplug.SWITCH_STATE_ON
+                if self.smartplug.context is None:
+                    self._device_id = self.smartplug.sys_info["deviceId"]
+                else:
+                    self._device_id = self.smartplug.context
+
+            if self.smartplug.context is None:
+                self._state = self.smartplug.state == self.smartplug.SWITCH_STATE_ON
+            else:
+                self._state = (
+                    list(
+                        filter(
+                            lambda child: child["id"] == self.smartplug.context,
```

Comment:
Please use a list comprehension instead of filter + lambda.

Reply:
@mattkasa FWIW the following is working with this branch installed as a `custom_component`:

```python
                self._state = [
                    child
                    for child in self.smartplug.sys_info["children"]
                    if child["id"] == self.smartplug.context
                ][0]["state"] == 1
```

Reply:
Thank you for this @shbatm! I've incorporated this suggestion 👍